scipy/stats/tests/test_generation/reference_distributions.py

```diff
@@ -0,0 +1,286 @@
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def logsf(self, x, dtype=np.float64):
+        fun = np.vectorize(self._logsf)
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def support(self, dtype=np.float64):
+        fun = np.vectorize(self._support)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def entropy(self, dtype=np.float64):
+        fun = np.vectorize(self._entropy)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def mean(self, dtype=np.float64):
+        fun = np.vectorize(self._mean)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def var(self, dtype=np.float64):
+        fun = np.vectorize(self._var)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def skew(self, dtype=np.float64):
+        fun = np.vectorize(self._skew)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def kurtosis(self, dtype=np.float64):
+        fun = np.vectorize(self._kurtosis)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+    def moment(self, order, center=None, dtype=np.float64):
+        fun = np.vectorize(self._moment)
+        order = self._make_mpf_array(order)
+        res = fun(order, **self._params)
+        return np.asarray(res, dtype=dtype)[()]
+
+
+class SkewNormal(ReferenceDistribution):
+    """Reference implementation of the SkewNormal distribution.
+
+    Follow the example here to generate new reference distributions.
+    Use the reference distributions to generate reference values of
+    distributions functions. For now, copy-paste the output into unit
+    tests. Full code to generate reference values does not need to be
+    included as a comment in the test; just refer to the reference
+    distribution used and the settings (e.g. mp.dps=50).
+    """
+
+    def __init__(self, *, a):
+        # Overriding __init__ is not necessary, but it allows IDEs to hint at
+        # shape parameters. All parameters are keyword only to the ambiguity
+        # inherent in positional arguments. The infrastructure does not take
+        # care of location and scale; nonetheless, assume standard location
+        # and scale. Typically, there is no need to test the SciPy distribution
+        # infrastructure's treatment of location and scale separately for
+        # a specific distribution.
+        super().__init__(a=a)
+
+    def _pdf(self, x, a):
+        # Write PDFs in the simplest form possible or following a (solid)
+        # reference as closely as possible. Trust mpmath for the accuracy, and
+        # don't worry about speed. What's important is the ease of verifying
+        # the PDF against the reference.
```

Comment:
If the result is inaccurate, it will not match SciPy's  output (whether SciPy is accurate or not). If this occurs,  try increasing dps before implementing a numerically favorable (but presumably more complex) implementation.
I should write something about this here.